### PR TITLE
add script to enhance sitemap with ds-view URLs

### LINKED ISSUE #378: Develop script to enhance sitemap with data set landing page URLs
## 💡 Description

Once https://github.com/NASA-PDS/portal-tasks/issues/65 is completed, we should augment the output file with URLs for all PDS data set landing pages

Super simple script to paginate through all the results from https://pds.nasa.gov/services/search/search?wt=json&q=product_class:Product_Collection%20OR%20product_class:Product_Bundle%20OR%20product_class:Product_Data_Set_PDS3%20OR%20product_class:Product_Document&fl=resLocation,modification_date&start=0&rows=1000

* Query URL
* Use `numFound` to keep track of total number of results
* Increment `start` and `rows` to page through the results
* Use `resLocation` for URL to data, use the latest `modification_date` (can be more than 1, and will require conversion to YYYY-MM-DD)
* Output should build an XML document that looks something like the following, which will be inserted into the sitemap:

```
  <url>
    <loc>https://pds.nasa.gov/ds-view/pds/viewDocument.jsp?identifier=urn%3Anasa%3Apds%3Asystem_bundle%3Adocument_pds4_standards%3Adph_1.17.0&version=1.0</loc>
    <lastmod>2021-10-14</lastmod>
  </url>
  <url>
    <loc>https://pds.nasa.gov/ds-view/pds/viewDocument.jsp?identifier=urn%3Anasa%3Apds%3Amisc%3Adocument_cassini%3Apds3_titan_ion_ug_itar_mar2012&version=1.0</loc>
    <lastmod>2021-04-20</lastmod>
  </url>
  ...
```

### PR DESCRIPTION
**Pull request by c-suh**
## 🗒️ Summary
Already reviewed and tested:
- Query URL
- Use `numFound` to keep track of total number of results
- Increment start and rows to page through the results
- Use `resLocation` for URL to data
- Use the latest `modification_date` (can be more than 1, and will require conversion to YYYY-MM-DD)
- Build an XML document that is inserted into the sitemap
- Git add, commit, and push the updated sitemap.xml to the repository for the PDS website

**Latest modifications:**
- use environment variable for file path
- dispose of separate virtual environment
- move files into this repository
- change name of main python file
- change name and location of log
- change time of cronjob to line up with other weekly tasks
- move back to other pds user

## ♻️ Related Issues
Resolves #378

### CODE REVIEW

File: bin/sitemap/ds-view.py [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+increment_slash_number_of_rows = 1000
+total_number_of_docs = None
+parsed_response = {}
+repo_path = os.environ.get('HTDOCS_HOME')
+target = os.path.join(repo_path, 'sitemap.xml')
+_logger = logging.getLogger(__name__)
+
+
+def do_git_stuff():
+    """Move the new sitemap to the repo. Update timestamp of local, partial sitemap. (In that order.)
```
[the remark this answers is not in the document]
Comment (nutjob4life) [reply bin/sitemap/ds-view.py:24]: @sonatype-lift ignore

File: bin/sitemap/ds-view.py [hunk 2]
```diff
@@ -0,0 +1,316 @@
+# https://github.com/NASA-PDS/operations/issues/378
+# enhance PDS sitemap with data set landing page URLs (results for ds-view)
+# written May 2023 by csuh (with minor contributions by kelly)
+from datetime import datetime
+from git import Repo
```
[the remark this answers is not in the document]
Comment (nutjob4life) [reply bin/sitemap/ds-view.py:5]: @sonatype-lift ignore

**Review by nutjob4life — APPROVED** [review]
Looks great! 👍

**Comment by c-suh** [comment]
@nutjob4life I might ask for a re-review in a bit. The script runs manually but errors when run by cron. Will get back to you.

**Comment by c-suh** [comment]
@nutjob4life never mind! It was rectified by a comment you made in the Slack discussion. However, these sonatype-lift checks are new to me. Did these become a part of our procedures somehow and I completely missed it?

**Comment by nutjob4life** [comment]
@c-suh yeah, SonaType Lift … it's an extra level of sanity checking that sometimes is more trouble than it's worth—not to mention that it's also going to be retired soon enough. You can decide if you want to make all of its warnings go away.

For example, it says there's no `Repo` in the `git` module in `from git import Repo`—but that's obviously wrong. But it also warns that there's a `write(f'   </url>\n')` that uses an f-string but has no placeholders—which is correct. That should just be `write('   </url>\n')`.

**Comment by c-suh** [comment]
@nutjob4life got it; thank you - both for the information and your help overall with this task!

**Pull request merged**